### PR TITLE
Add select API to torchx

### PR DESCRIPTION
**Pull request by howard0su**

### CODE REVIEW

File: torchx/lib/torchx/backend.ex [hunk 1]
```diff
@@ -748,6 +748,48 @@ defmodule Torchx.Backend do
...
+      |> from_nx()
+
+    on_false_torch =
+      on_false
+      |> from_nx()
+
+    if out_type == :bool do
+      pred_torch =
+        pred
+        |> from_nx()
```
Comment (josevalim) [inline torchx/lib/torchx/backend.ex:779]: Minor nitpick: we usually avoid `|>` if there is only one `|>`:

```suggestion
    on_true_torch = from_nx(on_true)
    on_false_torch = from_nx(on_false)

    if out_type == :bool do
      pred_torch = from_nx(pred)
```

:)

File: torchx/c_src/torchx.cpp [hunk 2]
```diff
@@ -608,6 +609,15 @@ NIF(clip)
   TENSOR(torch::clip(*t, *min, *max));
 }
 
+NIF(where)
+{
+  TENSOR_PARAM(0, cond);
+  TENSOR_PARAM(1, self);
+  TENSOR_PARAM(2, other);
+
+  TENSOR(torch::where(*cond, *self, *other));
```
Comment (polvalente) [inline torchx/c_src/torchx.cpp:618]:
```suggestion
  TENSOR_PARAM(1, on_true);
  TENSOR_PARAM(2, on_false);

  TENSOR(torch::where(*cond, *on_true, *on_false));
```

File: torchx/lib/torchx/backend.ex [hunk 3]
```diff
@@ -748,6 +748,48 @@ defmodule Torchx.Backend do
...
+    on_false_torch =
+      on_false
+      |> from_nx()
+
+    if out_type == {:u, 8} do
+      pred_torch =
+        pred
+        |> from_nx()
+
+      to_nx(Torchx.where(pred_torch, on_true_torch, on_false_torch), out)
```
Comment (polvalente) [inline torchx/lib/torchx/backend.ex:781]: On the other hand, I would have written this as a pipe haha

File: torchx/lib/torchx/backend.ex [hunk 4]
```diff
@@ -748,6 +748,48 @@ defmodule Torchx.Backend do
...
+    else
+      invert_pred_torch =
+        pred
+        |> from_nx()
+        |> Torchx.logical_not()
+
+      # swap true/false tensor
+      to_nx(Torchx.where(invert_pred_torch, on_false_torch, on_true_torch), out)
```
Comment (polvalente) [inline torchx/lib/torchx/backend.ex:789]: I didn't quite get this if/else. Could you explain?

Comment (josevalim) [reply torchx/lib/torchx/backend.ex:789]: We need to convert stuff to boolean and logical_not does it. But we need to swap the predicates!

Comment (polvalente) [reply torchx/lib/torchx/backend.ex:789]: Perhaps this could be a comment in the code

Comment (josevalim) [inline torchx/lib/torchx/backend.ex:789]:
```suggestion
    on_true_torch = from_nx(on_true)
    on_false_torch = from_nx(on_false)

    if out_type == {:u, 8} do
      pred
      |> from_nx()
      |> Torchx.where(on_true_torch, on_false_torch)
      |> to_nx(out)
    else
      pred
      |> from_nx()
      |> Torchx.logical_not()
      |> Torchx.where(on_false_torch, on_true_torch)
      |> to_nx(out)
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**